dags/common/utils.py

```diff
@@ -59,3 +61,16 @@ def parse_to_int(value):
         return int(value)
     except (ValueError, TypeError):
         logger.error("Cannot parse to integer", value=value)
+
+
+def preserve_cdata(article: str):
+    matches = CDATA_PATTERN.finditer(article)
+    for match in matches:
+        article = CDATA_PATTERN.sub(
+            match.group(1).replace("\\", "\\\\"), article, count=1
```

Comment:
why you need to pass `count`?

Reply:
because I want to change just the first occurrence of the CDATA and remove the CDATA tag, but leave the value inside of it. The values are different, but the pattern matches all CDATAs. Let's say, we have:
 <?CDATA $ \pm $?>
<?CDATA $(3.44 \pm 1.86_{\rm stat.}\pm $?>
<?CDATA $ 0.76_{\rm syst.})\times 10^{-4}$?>

So, I am taking the first CDATA and leaving just the value:
  $ \pm $?
<?CDATA $(3.44 \pm 1.86_{\rm stat.}\pm $?>
<?CDATA $ 0.76_{\rm syst.})\times 10^{-4}$?>

And later same, taking the first one and keeping just the value:
 $ \pm $?
$(3.44 \pm 1.86_{\rm stat.}\pm $
<?CDATA $ 0.76_{\rm syst.})\times 10^{-4}$?>
...

Without count, it will take all CDATAs and change to the first value from the matches, so in the end, I would have:
from this :
 <?CDATA $ \pm $?>
<?CDATA $(3.44 \pm 1.86_{\rm stat.}\pm $?>
<?CDATA $ 0.76_{\rm syst.})\times 10^{-4}$?>

To this:
 $ \pm $
 $ \pm $
 $ \pm $

while the correct one is:
$ \pm $?
$(3.44 \pm 1.86_{\rm stat.}\pm $
$ 0.76_{\rm syst.})\times 10^{-4}$

Reply:
Ok, I see. Maybe it would be possible to write a regex with [lookahead and lookbehind](https://stackoverflow.com/questions/2973436/regex-lookahead-lookbehind-and-atomic-groups) so that you'll get access directly to the match group that interests you?

Reply:
I need all of them, and I am getting all of them here. Just I need to change them accordingly to their value, which I don't know in advance

Reply:
I am actually getting all groups that interest me. The problem is to sub them, that's why I am using count